### PR TITLE
Bind instance role server directly to 169.254.169.254:80

### PR DESCRIPTION
**Pull request by lox**
This avoids the compatibility issues associated with #41 by binding directly to the 
address that the aws sdk's look for with `aws-vault server` (as root) and proxying to 
a less privileged http server run by exec on `127.0.0.1:9099`. 

The exec command tries to do this, including adding the alias on `lo` for `169.254.169.254` and falls back to using the env vars if not possible.

This approach is possible, but will require some change for Windows.

The major caveat is that this doesn't support parallel `aws-vault exec` invocations. The second one will fail to bind and will fall back to env.

### CODE REVIEW

**Comment by lox** [comment]
My take is that this is actually less voodoo than the current approach. The only bit that makes me uncomfortable is the exec command launching `aws-vault server` as a background process if it's not running, but I think it's still reasonable.

**Comment by mtibben** [comment]
I would vote for env vars being the default. Needing `sudo` and changing network configuration seems heavy handed and unexpected for a credential manager

**Comment by lox** [comment]
@mtibben yeah, I think you are right.

File: server.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	log.Printf("Fetched credentials from %s", localServerUrl)
+
+	w.Header().Set("Content-Type", "text/plain")
+	w.WriteHeader(http.StatusOK)
+
+	io.Copy(w, resp.Body)
+}
+
+func installNetworkAlias() ([]byte, error) {
+	return exec.Command("ifconfig", "lo0", "alias", "169.254.169.254").CombinedOutput()
```
Comment (mtibben) [inline server.go]: does this need to be removed at some point e.g. when the server is stopped ?

Comment (lox) [inline server.go]: Not really, it doesn't do any harm and it's addition is idempotent.

**Pull request merged**